src/autopas/containers/verletClusterLists/traversals/VerletClustersColoringTraversal.h

```diff
@@ -0,0 +1,169 @@
+  typedef typename VerletClusterMaths::index_t index_t;
+  /**
+   * Each base step looks like this:
+   *    X C N  Colors:  1 2 3
+   *    N N N           4 5 6
+   * Where C is the current cell, N are the neighbor cells that is worked on, and X is not worked on. The neighbor list
+   * with newton 3 of the VerletClusterLists container is build in a way that the neighbor lists already contain only
+   * the neighbor clusters of these cells.
+   * @see VerletClusterLists::updateVerletLists(bool)
+   */
+  static constexpr std::array<unsigned long, 3> stride{3ul, 2ul, 1ul};
+
+ private:
+  /**
+   * Helper method to iterate over one color cell.
+   * @param xColorCell The x coordinate of the cell.
+   * @param yColorCell The y coordinate of the cell.
+   * @param zColorCell The z coordinate of the cell.
+   * @param gridsPerColoringCell The number of grids that every cell has in every dimension.
+   */
+  void processColorCell(unsigned long xColorCell, unsigned long yColorCell, unsigned long zColorCell,
+                        int gridsPerColoringCell);
+
+  /**
+   * Helper method to traverse two neighbor clusters.
+   * @param clusterStart The first cluster.
+   * @param neighborClusterStart The second cluster.
+   * @param clusterSize The size of the cluster.
+   */
+  void traverseClusterPair(Particle *clusterStart, Particle *neighborClusterStart, int clusterSize);
+
+ public:
+  /**
+   * Constructor of the VerletClustersTraversal.
+   * @param pairwiseFunctor The functor to use for the traveral.
+   */
+  explicit VerletClustersColoringTraversal(PairwiseFunctor *pairwiseFunctor)
+      : CBasedTraversal<ParticleCell, PairwiseFunctor, dataLayout, useNewton3>({0, 0, 0}, pairwiseFunctor, 0, {}),
+        _functor(pairwiseFunctor) {}
+
+  TraversalOption getTraversalType() const override { return TraversalOption::verletClustersColoring; }
+
+  DataLayoutOption getDataLayout() const override { return dataLayout; }
+  bool getUseNewton3() const override { return useNewton3; }
+  bool isApplicable() const override { return (dataLayout == DataLayoutOption::aos); }
+
+  void initTraversal() override {}
+
+  void endTraversal() override {}
+
+  void traverseParticlePairs() override {
+    auto &clusterList = *VerletClustersTraversalInterface<Particle>::_verletClusterLists;
+
+    double gridSideLength = clusterList.getGridSideLength();
+    double cutoff = clusterList.getCutoff();
+    double gridsPerColoringCell = std::ceil(cutoff / gridSideLength);
+    std::array<unsigned long, 3> coloringCellsPerDim{};
+    for (int i = 0; i < 3; i++) {
+      coloringCellsPerDim[i] =
+          static_cast<unsigned long>(std::ceil(clusterList.getCellsPerDimension()[i] / gridsPerColoringCell));
+    }
+
+    auto loopBody = [this, gridsPerColoringCell](unsigned long x, unsigned long y, unsigned long z) {
+      processColorCell(x, y, z, gridsPerColoringCell);
+    };
+
+    const auto end = ArrayMath::sub(coloringCellsPerDim, {0ul, 0ul, 0ul});
```

Comment:
True, the subtractions comes from some legacy code, too. `coloringCellsPerDim[2]` should be 1 anyways. If it`s not, there is an error in calculatating `coloringCellsPerDim`, so that is why the exception is there below to check that.